src/build/tenscript/parser.rs

```diff
@@ -42,76 +40,62 @@ pub fn parse(source: &str) -> Result<FabricPlan, Error> {
         .map_err(|kind| Error::ParseError(ParseError { kind }))
 }
 
-macro_rules! expect_enum {
-        ($value:expr, { $($name:pat => $enum_val:expr,)+ }) => {
-            {
-                let expected = stringify!($($name)|+);
-                let $crate::build::tenscript::expression::Expression::Atom(ref name) = $value else {
-                    return Err($crate::build::tenscript::parser::ErrorKind::TypeError { expected, expression: $value.clone() })
-                };
-                match name.as_str() {
-                    $(
-                        $name => $enum_val,
-                    )+
-                    _ => return Err($crate::build::tenscript::parser::ErrorKind::TypeError { expected, expression: $value.clone() })
-                }
-            }
-        }
-    }
+// macro_rules! expect_enum {
+//         ($value:expr, { $($name:pat => $enum_val:expr,)+ }) => {
+//             {
+//                 let expected = stringify!($($name)|+);
+//                 let $crate::build::tenscript::expression::Expression::Atom(ref name) = $value else {
+//                     return Err($crate::build::tenscript::parser::ErrorKind::TypeError { expected, expression: $value.clone() })
+//                 };
+//                 match name.as_str() {
+//                     $(
+//                         $name => $enum_val,
+//                     )+
+//                     _ => return Err($crate::build::tenscript::parser::ErrorKind::TypeError { expected, expression: $value.clone() })
+//                 }
+//             }
+//         }
+//     }
 
 struct Call<'a> {
     head: &'a str,
     tail: &'a [Expression],
 }
 
-fn expect_call<'a>(rule: &'static str, expression: &'a Expression) -> Result<Call<'a>, ErrorKind> {
+fn expect_call(expression: &Expression) -> Result<Call, ErrorKind> {
```

Comment:
You don't want to pass in the rule anymore? I've always found that quite useful in debugging parser errors

Reply:
Something about the static lifetime kept causing a mild-disapproval.

Reply:
Weird haha